constantine/ethereum_eip7594_kzg.nim

```diff
@@ -0,0 +1,196 @@
+    let status = evalExpr # Ensure single evaluation
+    case status
+    of cttCodecEcc_Success:                             discard
+    of cttCodecEcc_InvalidEncoding:                     return cttEthKzg_EccInvalidEncoding
+    of cttCodecEcc_CoordinateGreaterThanOrEqualModulus: return cttEthKzg_EccCoordinateGreaterThanOrEqualModulus
+    of cttCodecEcc_PointNotOnCurve:                     return cttEthKzg_EccPointNotOnCurve
+    of cttCodecEcc_PointNotInSubgroup:                  return cttEthKzg_EccPointNotInSubGroup
+    of cttCodecEcc_PointAtInfinity:                     discard
+
+template check(Section: untyped, evalExpr: CttCodecScalarStatus): untyped {.dirty.} =
+  # Translate codec status code to KZG status code
+  # Exit current code block
+  block:
+    let status = evalExpr # Ensure single evaluation
+    case status
+    of cttCodecScalar_Success:                          discard
+    of cttCodecScalar_Zero:                             discard
+    of cttCodecScalar_ScalarLargerThanCurveOrder:       result = cttEthKzg_ScalarLargerThanCurveOrder; break Section
+
+template check(Section: untyped, evalExpr: CttCodecEccStatus): untyped {.dirty.} =
+  # Translate codec status code to KZG status code
+  # Exit current code block
+  block:
+    let status = evalExpr # Ensure single evaluation
+    case status
+    of cttCodecEcc_Success:                             discard
+    of cttCodecEcc_InvalidEncoding:                     result = cttEthKzg_EccInvalidEncoding; break Section
+    of cttCodecEcc_CoordinateGreaterThanOrEqualModulus: result = cttEthKzg_EccCoordinateGreaterThanOrEqualModulus; break Section
+    of cttCodecEcc_PointNotOnCurve:                     result = cttEthKzg_EccPointNotOnCurve; break Section
+    of cttCodecEcc_PointNotInSubgroup:                  result = cttEthKzg_EccPointNotInSubGroup; break Section
+    of cttCodecEcc_PointAtInfinity:                     discard
+
+
+# from eip4844, import
+# bytes_to_bls_field
+# bls_field_to_bytes
+
+func cell_to_coset_evals(evals: var CosetEvals,cell: Cell): CttCodecEccStatus=
+  # Convert an untrusted ``Cell`` into a trusted ``CosetEvals``
+  var 
+    start: int
+    ending: int
+    value:Fr[BLS12_381]
+
+  let view = cast[ptr array[FIELD_ELEMENTS_PER_CELL, array[32, byte]]](cell.unsafeAddr)
+  for i in 0..FIELD_ELEMENTS_PER_CELL:
+    start = i*BYTES_PER_FIELD_ELEMENT
+    ending = (i+1)*BYTES_PER_FIELD_ELEMENT
+    var status= value.bytes_to_bls_field(view[i])
+    if status notin {cttCodecScalar_Success, cttCodecScalar_Zero}:
+      return cttCodecEcc_PointNotOnCurve
+    evals[i]=value
+
+  return cttCodecEcc_Success
+
+
+
+func coset_evals_to_cell(dst: var Cell,coset_evals: CosetEvals): CttCodecEccStatus=
+    # Convert a trusted ``CosetEval`` into an untrusted ``Cell``.
+    # for i in 0 ..< FIELD_ELEMENTS_PER_CELL:
+    #     var bytes = bls_field_to_bytes(cosetEvals[i])
+    #     for j in 0 ..< bytes.len:
+    #         dst[i* bytes.len + j] = bytes[j]
```

Comment:
```suggestion
  #         dst[i* bytes.len + j] = bytes[j]
```

2 spaces everywhere, you can configure vs-code with 2 spaces for Nim with the following

ctrl+shift+P: Open users settings (JSON)

and add a language specific config
```json
    "[nim]": {
        "editor.tabSize": 2,
    },
```

like this:
![image](https://github.com/user-attachments/assets/c27e8d4e-efd5-4e89-9b5b-b54f7546e1b5)